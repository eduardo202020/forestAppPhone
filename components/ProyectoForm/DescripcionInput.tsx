import React from 'react';
import { TextInput, Text, StyleSheet } from 'react-native';

interface DescripcionInputProps {
  value: string;
  onChange: (v: string) => void;
  editable?: boolean;
  error?: string;
}

const DescripcionInput: React.FC<DescripcionInputProps> = ({
  value,
  onChange,
  editable = true,
  error,
}) => (
  <>
    <TextInput
      style={[styles.input, { minHeight: 60 }]}
      placeholder="DescripciÃ³n"
      value={value}
      onChangeText={onChange}
      multiline
      editable={editable}
    />
    {error && <Text style={styles.errorText}>{error}</Text>}
  </>
);

const styles = StyleSheet.create({
  input: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 8,
    padding: 10,
    marginBottom: 12,
    fontSize: 16,
    backgroundColor: '#f9f9f9',
  },
  errorText: {
    color: 'red',
    marginBottom: 8,
    fontSize: 13,
  },
});

export default DescripcionInput;
