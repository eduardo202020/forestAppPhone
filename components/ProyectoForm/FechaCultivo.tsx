import React from 'react';
import { Text, StyleSheet } from 'react-native';
import { Calendar } from 'react-native-calendars';

interface FechaCultivoProps {
  selectedDate: string;
  setSelectedDate: (v: string) => void;
  error?: string;
}

const FechaCultivo: React.FC<FechaCultivoProps> = ({
  selectedDate,
  setSelectedDate,
  error,
}) => (
  <>
    <Text style={styles.label}>Fecha de Cultivo</Text>
    <Calendar
      onDayPress={(day: { dateString: string }) =>
        setSelectedDate(day.dateString)
      }
      markedDates={
        selectedDate
          ? { [selectedDate]: { selected: true, selectedColor: '#2ecc71' } }
          : {}
      }
      current={selectedDate}
      style={{ marginBottom: 16 }}
    />
    <Text style={styles.ubicacionText}>
      {selectedDate ? `Seleccionada: ${selectedDate}` : 'Selecciona una fecha'}
    </Text>
    {error && <Text style={styles.errorText}>{error}</Text>}
  </>
);

const styles = StyleSheet.create({
  label: {
    fontWeight: 'bold',
    marginBottom: 4,
    marginTop: 8,
  },
  ubicacionText: {
    fontSize: 12,
    color: '#888',
    marginBottom: 10,
  },
  errorText: {
    color: 'red',
    marginBottom: 8,
    fontSize: 13,
  },
});

export default FechaCultivo;
