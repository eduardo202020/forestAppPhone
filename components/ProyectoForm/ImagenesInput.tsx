import React from 'react';
import {
  View,
  Pressable,
  Text,
  ScrollView,
  Image,
  StyleSheet,
  Modal,
} from 'react-native';

interface ImagenesInputProps {
  imagenes: string[];
  setModalVisible: (v: boolean) => void;
  modalVisible: boolean;
  horizontal?: boolean;
}

const ImagenesInput: React.FC<ImagenesInputProps> = ({
  imagenes,
  setModalVisible,
  modalVisible,
  horizontal = false,
}) => (
  <>
    <Text style={styles.label}>Imágenes</Text>
    {horizontal ? (
      <ScrollView horizontal style={styles.imagesRow}>
        {imagenes.map((img, idx) => (
          <Pressable
            key={idx}
            style={styles.imageBox}
            onPress={() => setModalVisible(true)}
          >
            <Image
              source={{ uri: img }}
              style={{ width: '100%', height: '100%', borderRadius: 10 }}
            />
          </Pressable>
        ))}
      </ScrollView>
    ) : (
      <View style={styles.imagesRow}>
        {imagenes.map((img, idx) => (
          <Pressable
            key={idx}
            style={styles.imageBox}
            onPress={() => setModalVisible(true)}
          >
            <Text style={{ color: '#888', textAlign: 'center' }}>
              Imagen {idx + 1}
            </Text>
          </Pressable>
        ))}
      </View>
    )}
    <Modal
      visible={modalVisible}
      transparent
      animationType="fade"
      onRequestClose={() => setModalVisible(false)}
    >
      <View style={styles.modalContainer}>
        <View style={styles.modalContent}>
          <Text>Funcionalidad para añadir imágenes próximamente</Text>
          <Text style={{ marginTop: 10, color: '#888' }}>
            (Próximamente podrás añadir o editar imágenes)
          </Text>
        </View>
      </View>
    </Modal>
  </>
);

const styles = StyleSheet.create({
  label: {
    fontWeight: 'bold',
    marginBottom: 4,
    marginTop: 8,
  },
  imagesRow: {
    flexDirection: 'row',
    gap: 8,
    marginBottom: 16,
  },
  imageBox: {
    width: 100,
    height: 100,
    backgroundColor: '#e0e0e0',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 8,
    marginBottom: 16,
  },
  modalContainer: {
    flex: 1,
    backgroundColor: 'rgba(0,0,0,0.3)',
    alignItems: 'center',
    justifyContent: 'center',
  },
  modalContent: {
    backgroundColor: '#fff',
    padding: 24,
    borderRadius: 10,
    alignItems: 'center',
  },
});

export default ImagenesInput;
