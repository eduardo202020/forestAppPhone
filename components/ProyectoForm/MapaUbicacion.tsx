import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import MapView, { Marker } from 'react-native-maps';

interface MapaUbicacionProps {
  selectedLocation: { latitude: number; longitude: number } | null;
  mapRef?: React.RefObject<MapView>;
  onMapPress?: (e: any) => void;
  editable?: boolean;
  error?: string;
}

const MapaUbicacion: React.FC<MapaUbicacionProps> = ({
  selectedLocation,
  mapRef,
  onMapPress,
  editable = true,
  error,
}) => (
  <>
    <Text style={styles.label}>Ubicación</Text>
    <MapView
      ref={mapRef}
      style={styles.map}
      initialRegion={{
        latitude: selectedLocation?.latitude || -12.0464,
        longitude: selectedLocation?.longitude || -77.0428,
        latitudeDelta: 0.0922,
        longitudeDelta: 0.0421,
      }}
      onPress={editable ? onMapPress : undefined}
    >
      {selectedLocation && <Marker coordinate={selectedLocation} />}
    </MapView>
    <Text style={styles.ubicacionText}>
      {selectedLocation
        ? `Ubicación seleccionada: ${selectedLocation.latitude.toFixed(5)}, ${selectedLocation.longitude.toFixed(5)}`
        : 'Toca el mapa para seleccionar una ubicación'}
    </Text>
    {error && <Text style={styles.errorText}>{error}</Text>}
  </>
);

const styles = StyleSheet.create({
  map: {
    width: '100%',
    height: 180,
    borderRadius: 10,
    marginBottom: 10,
  },
  ubicacionText: {
    fontSize: 12,
    color: '#888',
    marginBottom: 10,
  },
  label: {
    fontWeight: 'bold',
    marginBottom: 4,
    marginTop: 8,
  },
  errorText: {
    color: 'red',
    marginBottom: 8,
    fontSize: 13,
  },
});

export default MapaUbicacion;
