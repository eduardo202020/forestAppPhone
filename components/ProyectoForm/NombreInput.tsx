import React from 'react';
import { TextInput, Text, StyleSheet } from 'react-native';

interface NombreInputProps {
  value: string;
  onChange: (v: string) => void;
  editable?: boolean;
  error?: string;
}

const NombreInput: React.FC<NombreInputProps> = ({
  value,
  onChange,
  editable = true,
  error,
}) => (
  <>
    <TextInput
      style={styles.input}
      placeholder="Nombre del proyecto"
      value={value}
      onChangeText={onChange}
      editable={editable}
    />
    {error && <Text style={styles.errorText}>{error}</Text>}
  </>
);

const styles = StyleSheet.create({
  input: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 8,
    padding: 10,
    marginBottom: 12,
    fontSize: 16,
    backgroundColor: '#f9f9f9',
  },
  errorText: {
    color: 'red',
    marginBottom: 8,
    fontSize: 13,
  },
});

export default NombreInput;
