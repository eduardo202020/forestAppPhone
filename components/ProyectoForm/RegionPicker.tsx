import React from 'react';
import { Picker } from '@react-native-picker/picker';
import { Text, StyleSheet } from 'react-native';

interface RegionPickerProps {
  value: string;
  onChange: (v: string) => void;
  editable?: boolean;
  error?: string;
}

const RegionPicker: React.FC<RegionPickerProps> = ({
  value,
  onChange,
  editable = true,
  error,
}) => (
  <>
    <Picker
      selectedValue={value}
      onValueChange={onChange}
      style={styles.input}
      enabled={editable}
    >
      <Picker.Item label="Selecciona una región" value="" />
      <Picker.Item label="Loreto" value="Loreto" />
      <Picker.Item label="Cusco" value="Cusco" />
      <Picker.Item label="Puno" value="Puno" />
      <Picker.Item label="Madre de Dios" value="Madre de Dios" />
      <Picker.Item label="Amazonas" value="Amazonas" />
      <Picker.Item label="San Martín" value="San Martín" />
    </Picker>
    {error && <Text style={styles.errorText}>{error}</Text>}
  </>
);

const styles = StyleSheet.create({
  input: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 8,
    padding: 10,
    marginBottom: 12,
    fontSize: 16,
    backgroundColor: '#f9f9f9',
  },
  errorText: {
    color: 'red',
    marginBottom: 8,
    fontSize: 13,
  },
});

export default RegionPicker;
