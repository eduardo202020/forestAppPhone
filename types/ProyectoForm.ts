export interface ProyectoFormProps {
  nombre: string;
  setNombre: (_: string) => void;
  region: string;
  setRegion: (_: string) => void;
  descripcion: string;
  setDescripcion: (_: string) => void;
  selectedDate: string;
  setSelectedDate: (_: string) => void;
  selectedLocation: { latitude: number; longitude: number } | null;
  imagenes: string[];
  setModalVisible: (_: boolean) => void;
  modalVisible: boolean;
  errores?: { [key: string]: string };
}
