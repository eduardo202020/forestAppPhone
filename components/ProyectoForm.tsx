import React from 'react';
import { StyleSheet } from 'react-native';
import { Picker } from '@react-native-picker/picker';
import MapView, { Marker } from 'react-native-maps';
import { Calendar } from 'react-native-calendars';
import NombreInput from './ProyectoForm/NombreInput';
import RegionPicker from './ProyectoForm/RegionPicker';
import MapaUbicacion from './ProyectoForm/MapaUbicacion';
import FechaCultivo from './ProyectoForm/FechaCultivo';
import DescripcionInput from './ProyectoForm/DescripcionInput';
import ImagenesInput from './ProyectoForm/ImagenesInput';

interface ProyectoFormProps {
  nombre: string;
  setNombre: (v: string) => void;
  region: string;
  setRegion: (v: string) => void;
  descripcion: string;
  setDescripcion: (v: string) => void;
  selectedDate: string;
  setSelectedDate: (v: string) => void;
  selectedLocation: { latitude: number; longitude: number } | null;
  setSelectedLocation: (v: { latitude: number; longitude: number }) => void;
  ubicacion: string;
  setUbicacion: (v: string) => void;
  imagenes: string[];
  setModalVisible: (v: boolean) => void;
  modalVisible: boolean;
  errores?: { [key: string]: string };
  mapRef?: React.RefObject<MapView>;
  onMapPress?: (e: any) => void;
  onRegionChange?: (itemValue: string) => void;
  editable?: boolean;
  horizontalImages?: boolean;
}

const ProyectoForm: React.FC<ProyectoFormProps> = ({
  nombre,
  setNombre,
  region,
  setRegion,
  descripcion,
  setDescripcion,
  selectedDate,
  setSelectedDate,
  selectedLocation,
  setSelectedLocation,
  ubicacion,
  setUbicacion,
  imagenes,
  setModalVisible,
  modalVisible,
  errores = {},
  mapRef,
  onMapPress,
  onRegionChange,
  editable = true,
  horizontalImages = false,
}) => {
  return (
    <>
      <NombreInput
        value={nombre}
        onChange={setNombre}
        editable={editable}
        error={errores.nombre}
      />
      <RegionPicker
        value={region}
        onChange={onRegionChange || setRegion}
        editable={editable}
        error={errores.region}
      />
      <MapaUbicacion
        selectedLocation={selectedLocation}
        mapRef={mapRef}
        onMapPress={onMapPress}
        editable={editable}
        error={errores.ubicacion}
      />
      <FechaCultivo
        selectedDate={selectedDate}
        setSelectedDate={setSelectedDate}
        error={errores.fecha}
      />
      <DescripcionInput
        value={descripcion}
        onChange={setDescripcion}
        editable={editable}
        error={errores.descripcion}
      />
      <ImagenesInput
        imagenes={imagenes}
        setModalVisible={setModalVisible}
        modalVisible={modalVisible}
        horizontal={horizontalImages}
      />
    </>
  );
};

const styles = StyleSheet.create({
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 16,
  },
  input: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 8,
    padding: 10,
    marginBottom: 12,
    fontSize: 16,
    backgroundColor: '#f9f9f9',
  },
  map: {
    width: '100%',
    height: 180,
    borderRadius: 10,
    marginBottom: 10,
  },
  ubicacionText: {
    fontSize: 12,
    color: '#888',
    marginBottom: 10,
  },
  imageBox: {
    width: 100,
    height: 100,
    backgroundColor: '#e0e0e0',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 8,
    marginBottom: 16,
  },
  label: {
    fontWeight: 'bold',
    marginBottom: 4,
    marginTop: 8,
  },
  modalContainer: {
    flex: 1,
    backgroundColor: 'rgba(0,0,0,0.3)',
    alignItems: 'center',
    justifyContent: 'center',
  },
  modalContent: {
    backgroundColor: '#fff',
    padding: 24,
    borderRadius: 10,
    alignItems: 'center',
  },
  imagesRow: {
    flexDirection: 'row',
    gap: 8,
    marginBottom: 16,
  },
  errorText: {
    color: 'red',
    marginBottom: 8,
    fontSize: 13,
  },
});

export default ProyectoForm;
