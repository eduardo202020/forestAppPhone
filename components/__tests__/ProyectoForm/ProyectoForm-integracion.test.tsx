import React from 'react';
import renderer from 'react-test-renderer';
import ProyectoForm from '../../ProyectoForm';

jest.mock('react-native/Libraries/Components/Pressable/Pressable', () => {
  const React = require('react');
  return {
    __esModule: true,
    default: (props: any) => <pressable-mock {...props} />,
  };
});
jest.mock('@react-native-picker/picker', () => {
  const React = require('react');
  const Picker = (props: any) => <picker-mock {...props} />;
  Picker.Item = (props: any) => <picker-item-mock {...props} />;
  return { Picker };
});
jest.mock('react-native-calendars', () => {
  const React = require('react');
  return {
    Calendar: (props: any) => <calendar-mock {...props} />,
  };
});
jest.mock('react-native-maps', () => {
  const React = require('react');
  return {
    __esModule: true,
    default: (props: any) => <mapview-mock {...props} />,
  };
});

declare global {
  namespace JSX {
    interface IntrinsicElements {
      'picker-mock': any;
      'picker-item-mock': any;
      'calendar-mock': any;
      'mapview-mock': any;
      'pressable-mock': any;
    }
  }
}

// Mocks mínimos para simular el formulario completo
const defaultProps = {
  nombre: '',
  setNombre: jest.fn(),
  region: '',
  setRegion: jest.fn(),
  descripcion: '',
  setDescripcion: jest.fn(),
  selectedDate: '',
  setSelectedDate: jest.fn(),
  selectedLocation: null,
  setSelectedLocation: jest.fn(),
  ubicacion: '',
  setUbicacion: jest.fn(),
  imagenes: ['img1', 'img2'],
  setModalVisible: jest.fn(),
  modalVisible: false,
  errores: {},
  mapRef: undefined,
  onMapPress: jest.fn(),
  onRegionChange: jest.fn(),
  editable: true,
  horizontalImages: false,
};

describe('Integración ProyectoForm', () => {
  it('renderiza el formulario completo sin errores', () => {
    const tree = renderer.create(<ProyectoForm {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('muestra errores de validación en todos los campos', () => {
    const errores = {
      nombre: 'Falta nombre',
      region: 'Falta región',
      descripcion: 'Falta descripción',
      fecha: 'Falta fecha',
      ubicacion: 'Falta ubicación',
    };
    const tree = renderer
      .create(<ProyectoForm {...defaultProps} errores={errores} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('llama a los callbacks de los campos principales', () => {
    const setNombre = jest.fn();
    const setDescripcion = jest.fn();
    const setRegion = jest.fn();
    const setSelectedDate = jest.fn();
    const setSelectedLocation = jest.fn();
    const setUbicacion = jest.fn();
    const setModalVisible = jest.fn();
    const onMapPress = jest.fn();
    const onRegionChange = jest.fn();
    const testRenderer = renderer.create(
      <ProyectoForm
        {...defaultProps}
        setNombre={setNombre}
        setDescripcion={setDescripcion}
        setRegion={setRegion}
        setSelectedDate={setSelectedDate}
        setSelectedLocation={setSelectedLocation}
        setUbicacion={setUbicacion}
        setModalVisible={setModalVisible}
        onMapPress={onMapPress}
        onRegionChange={onRegionChange}
      />
    );
    // Simula cambios en los campos
    const root = testRenderer.root;
    root
      .findByProps({ placeholder: 'Nombre del proyecto' })
      .props.onChangeText('nuevo nombre');
    root.findByProps({ placeholder: 'Descripción' }).props.onChangeText('desc');
    // Simula selección de región
    root.findAllByType('picker-mock')[0].props.onValueChange('Loreto');
    // Simula selección de fecha
    root
      .findAllByType('calendar-mock')[0]
      .props.onDayPress({ dateString: '2025-05-17' });
    // Simula click en mapa
    if (root.findAllByType('mapview-mock')[0]) {
      root
        .findAllByType('mapview-mock')[0]
        .props.onPress({
          nativeEvent: { coordinate: { latitude: 1, longitude: 2 } },
        });
    }
    // Simula click en imagen
    root.findAllByType('pressable-mock')[0].props.onPress();
    // Verifica que los callbacks hayan sido llamados
    expect(setNombre).toHaveBeenCalledWith('nuevo nombre');
    expect(setDescripcion).toHaveBeenCalledWith('desc');
    expect(onRegionChange).toHaveBeenCalledWith('Loreto');
    expect(setSelectedDate).toHaveBeenCalledWith('2025-05-17');
    expect(onMapPress).toHaveBeenCalled();
    expect(setModalVisible).toHaveBeenCalled();
  });
});
