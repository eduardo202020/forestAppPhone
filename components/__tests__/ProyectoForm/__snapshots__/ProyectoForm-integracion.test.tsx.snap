// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integración ProyectoForm muestra errores de validación en todos los campos 1`] = `
[
  <TextInput
    editable={true}
    onChangeText={[MockFunction]}
    placeholder="Nombre del proyecto"
    style={
      {
        "backgroundColor": "#f9f9f9",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "fontSize": 16,
        "marginBottom": 12,
        "padding": 10,
      }
    }
    value=""
  />,
  <Text
    style={
      {
        "color": "red",
        "fontSize": 13,
        "marginBottom": 8,
      }
    }
  >
    Falta nombre
  </Text>,
  <picker-mock
    enabled={true}
    onValueChange={[MockFunction]}
    selectedValue=""
    style={
      {
        "backgroundColor": "#f9f9f9",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "fontSize": 16,
        "marginBottom": 12,
        "padding": 10,
      }
    }
  >
    <picker-item-mock
      label="Selecciona una región"
      value=""
    />
    <picker-item-mock
      label="Loreto"
      value="Loreto"
    />
    <picker-item-mock
      label="Cusco"
      value="Cusco"
    />
    <picker-item-mock
      label="Puno"
      value="Puno"
    />
    <picker-item-mock
      label="Madre de Dios"
      value="Madre de Dios"
    />
    <picker-item-mock
      label="Amazonas"
      value="Amazonas"
    />
    <picker-item-mock
      label="San Martín"
      value="San Martín"
    />
  </picker-mock>,
  <Text
    style={
      {
        "color": "red",
        "fontSize": 13,
        "marginBottom": 8,
      }
    }
  >
    Falta región
  </Text>,
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginBottom": 4,
        "marginTop": 8,
      }
    }
  >
    Ubicación
  </Text>,
  <mapview-mock
    initialRegion={
      {
        "latitude": -12.0464,
        "latitudeDelta": 0.0922,
        "longitude": -77.0428,
        "longitudeDelta": 0.0421,
      }
    }
    onPress={[MockFunction]}
    style={
      {
        "borderRadius": 10,
        "height": 180,
        "marginBottom": 10,
        "width": "100%",
      }
    }
  />,
  <Text
    style={
      {
        "color": "#888",
        "fontSize": 12,
        "marginBottom": 10,
      }
    }
  >
    Toca el mapa para seleccionar una ubicación
  </Text>,
  <Text
    style={
      {
        "color": "red",
        "fontSize": 13,
        "marginBottom": 8,
      }
    }
  >
    Falta ubicación
  </Text>,
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginBottom": 4,
        "marginTop": 8,
      }
    }
  >
    Fecha de Cultivo
  </Text>,
  <calendar-mock
    current=""
    markedDates={{}}
    onDayPress={[Function]}
    style={
      {
        "marginBottom": 16,
      }
    }
  />,
  <Text
    style={
      {
        "color": "#888",
        "fontSize": 12,
        "marginBottom": 10,
      }
    }
  >
    Selecciona una fecha
  </Text>,
  <Text
    style={
      {
        "color": "red",
        "fontSize": 13,
        "marginBottom": 8,
      }
    }
  >
    Falta fecha
  </Text>,
  <TextInput
    editable={true}
    multiline={true}
    onChangeText={[MockFunction]}
    placeholder="Descripción"
    style={
      [
        {
          "backgroundColor": "#f9f9f9",
          "borderColor": "#ccc",
          "borderRadius": 8,
          "borderWidth": 1,
          "fontSize": 16,
          "marginBottom": 12,
          "padding": 10,
        },
        {
          "minHeight": 60,
        },
      ]
    }
    value=""
  />,
  <Text
    style={
      {
        "color": "red",
        "fontSize": 13,
        "marginBottom": 8,
      }
    }
  >
    Falta descripción
  </Text>,
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginBottom": 4,
        "marginTop": 8,
      }
    }
  >
    Imágenes
  </Text>,
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 8,
        "marginBottom": 16,
      }
    }
  >
    <pressable-mock
      onPress={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e0e0e0",
          "borderRadius": 10,
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 16,
          "marginRight": 8,
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "#888",
            "textAlign": "center",
          }
        }
      >
        Imagen 
        1
      </Text>
    </pressable-mock>
    <pressable-mock
      onPress={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e0e0e0",
          "borderRadius": 10,
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 16,
          "marginRight": 8,
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "#888",
            "textAlign": "center",
          }
        }
      >
        Imagen 
        2
      </Text>
    </pressable-mock>
  </View>,
]
`;

exports[`Integración ProyectoForm renderiza el formulario completo sin errores 1`] = `
[
  <TextInput
    editable={true}
    onChangeText={[MockFunction]}
    placeholder="Nombre del proyecto"
    style={
      {
        "backgroundColor": "#f9f9f9",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "fontSize": 16,
        "marginBottom": 12,
        "padding": 10,
      }
    }
    value=""
  />,
  <picker-mock
    enabled={true}
    onValueChange={[MockFunction]}
    selectedValue=""
    style={
      {
        "backgroundColor": "#f9f9f9",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "fontSize": 16,
        "marginBottom": 12,
        "padding": 10,
      }
    }
  >
    <picker-item-mock
      label="Selecciona una región"
      value=""
    />
    <picker-item-mock
      label="Loreto"
      value="Loreto"
    />
    <picker-item-mock
      label="Cusco"
      value="Cusco"
    />
    <picker-item-mock
      label="Puno"
      value="Puno"
    />
    <picker-item-mock
      label="Madre de Dios"
      value="Madre de Dios"
    />
    <picker-item-mock
      label="Amazonas"
      value="Amazonas"
    />
    <picker-item-mock
      label="San Martín"
      value="San Martín"
    />
  </picker-mock>,
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginBottom": 4,
        "marginTop": 8,
      }
    }
  >
    Ubicación
  </Text>,
  <mapview-mock
    initialRegion={
      {
        "latitude": -12.0464,
        "latitudeDelta": 0.0922,
        "longitude": -77.0428,
        "longitudeDelta": 0.0421,
      }
    }
    onPress={[MockFunction]}
    style={
      {
        "borderRadius": 10,
        "height": 180,
        "marginBottom": 10,
        "width": "100%",
      }
    }
  />,
  <Text
    style={
      {
        "color": "#888",
        "fontSize": 12,
        "marginBottom": 10,
      }
    }
  >
    Toca el mapa para seleccionar una ubicación
  </Text>,
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginBottom": 4,
        "marginTop": 8,
      }
    }
  >
    Fecha de Cultivo
  </Text>,
  <calendar-mock
    current=""
    markedDates={{}}
    onDayPress={[Function]}
    style={
      {
        "marginBottom": 16,
      }
    }
  />,
  <Text
    style={
      {
        "color": "#888",
        "fontSize": 12,
        "marginBottom": 10,
      }
    }
  >
    Selecciona una fecha
  </Text>,
  <TextInput
    editable={true}
    multiline={true}
    onChangeText={[MockFunction]}
    placeholder="Descripción"
    style={
      [
        {
          "backgroundColor": "#f9f9f9",
          "borderColor": "#ccc",
          "borderRadius": 8,
          "borderWidth": 1,
          "fontSize": 16,
          "marginBottom": 12,
          "padding": 10,
        },
        {
          "minHeight": 60,
        },
      ]
    }
    value=""
  />,
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginBottom": 4,
        "marginTop": 8,
      }
    }
  >
    Imágenes
  </Text>,
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 8,
        "marginBottom": 16,
      }
    }
  >
    <pressable-mock
      onPress={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e0e0e0",
          "borderRadius": 10,
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 16,
          "marginRight": 8,
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "#888",
            "textAlign": "center",
          }
        }
      >
        Imagen 
        1
      </Text>
    </pressable-mock>
    <pressable-mock
      onPress={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e0e0e0",
          "borderRadius": 10,
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 16,
          "marginRight": 8,
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "#888",
            "textAlign": "center",
          }
        }
      >
        Imagen 
        2
      </Text>
    </pressable-mock>
  </View>,
]
`;
